src/catalog/catalog_entry/table_catalog_entry.cpp

```diff
@@ -133,9 +155,17 @@ unique_ptr<CatalogEntry> TableCatalogEntry::AlterEntry(ClientContext &context, A
 	case AlterTableType::RENAME_TABLE: {
 		auto rename_info = (RenameTableInfo *)table_info;
 		auto copied_table = Copy(context);
+		auto &entry = (TableCatalogEntry &)*copied_table;
+		for (auto &gen_col : entry.generated_columns) {
+			gen_col.RenameTable(*rename_info);
```

Comment:
At first this was a hunch, but I tested and ambiguity did occur if the table name wasn't baked into the columnref
```sql
CREATE TABLE foo (
	var INTEGER,
	gen INTEGER GENERATED ALWAYS AS(var * 5)
);
CREATE TABLE bar (
	var INTEGER
);

SELECT gen FROM foo, bar;
```

Reply:
This should be resolved when the actual replacement/binding happens, no? Otherwise aliases would not work correctly either, e.g. this will probably fail:

```sql
CREATE TABLE foo (
	var INTEGER,
	gen INTEGER GENERATED ALWAYS AS(var * 5) VIRTUAL
);
SELECT gen FROM foo AS bar;
```

Reply:
Fixed now, it was indeed broken 👍